src/client/scripts/views/SignedIn/DiscoverView.js

```diff
@@ -157,10 +165,89 @@ export class DiscoverView {
      */
     async #renderFilterBar() { // TODO: Gauri
         const elm = document.createElement('div');
+        // elm.style.marginBottom = "100px";
 
         // TODO: Render the filter bar - should have checkboxes associated with the character property in the User data structure,
         // and just a few properties in the Preferences data structure (otherwise it's too much work). Only add the Preferences
         // properties if !this.#curUser.hasHousing
+        
+        // Dictionary for filtering users 
+        const filterOptions = {
+            "Gender": {
+                "female": false,
+                "male": false,
+                "nonbinary": false
+            },
+            "Clean": {
+                "messy": false,
+                "clean": false,
+                "very clean": false
+            },
+            "Sleep": {
+                "morning riser": false,
+                "afternoon riser": false,
+                "evening riser": false
+            },
+            "Noise": {
+                "low noise": false,
+                "medium noise": false,
+                "loud noise": false
+            },
+            "Guests": {
+                "no to less guests": false,
+                "some guests": false,
+                "any number of guests": false
+            },
+            "Education Level": {
+                "undergrad": false,
+                "postgrad": false,
+                "other": false
+            }
+        };
+
+        // Dictionary to filter users (based on housing preferences)
+        const preferenceOptions = {
+            "Rent": {
+                "minimum rent": false,
+                "maximum rent": false
+            },
+            "Occupants": {
+                "minimum occupants": false,
+                "maximum occupants": false
```

Comment:
The numerical input ones are gonna be annoying to implement so could you do any other property haha sorry

Reply:
No worries, I wasn't too sure about those either - I suppose room type and building type could replace these

Reply:
Yeah dw abt adding everything, just a few things. The application doesn't have to be completely polished, just enough to show a start.